src/libcore/ffi.rs

```diff
@@ -363,5 +401,64 @@ extern "rust-intrinsic" {
 
     /// Loads an argument of type `T` from the `va_list` `ap` and increment the
     /// argument `ap` points to.
+    #[cfg(not(all(target_arch = "aarch64", not(target_os = "ios"), not(windows))))]
     fn va_arg<T: sealed_trait::VaArgSafe>(ap: &mut VaListImpl<'_>) -> T;
 }
+
+#[cfg(all(target_arch = "aarch64", not(target_os = "ios"), not(windows)))]
+use crate::{mem, ptr};
+
+// Implemenation for AArch64 linux (and others that follow the standard PCS)
+// based on https://developer.arm.com/docs/ihi0055/d/procedure-call-standard-for-the-arm-64-bit-architecture
+// APPENDIX Variable argument Lists -> The va_start() macro
+#[cfg(all(target_arch = "aarch64", not(target_os = "ios"), not(windows)))]
+impl<'f> VaListImpl<'f> {
+    unsafe fn get_begin_and_end<T>(current_pos: usize, reg_size: usize) -> (usize, usize) {
+        let mut begin = current_pos;
+        if mem::align_of::<T>() > 8 && reg_size == 8 {
+            begin = begin.wrapping_add(15) & !15; // round up
+        }
+        let nreg = (mem::size_of::<T>() + reg_size - 1) / reg_size;
+        let end = begin.wrapping_add(nreg * reg_size);
+        #[cfg(target_endian = "big")]
+        {
+            if
+            /* classof(type) != "aggregate" && */
+            mem::size_of::<T>() < reg_size {
+                begin = begin.wrapping_add(reg_size - mem::size_of::<T>());
+            }
+        }
+        (begin, end)
+    }
+
+    unsafe fn va_arg_on_stack<T: sealed_trait::VaArgSafe>(stack: &mut *mut c_void) -> T {
+        let (begin, end) = Self::get_begin_and_end::<T>(*stack as usize, 8);
```

Comment:
AFAIK this can give us different values than the spec if the stack isn't 8 byte aligned, but I suppose much worse things would be happening if that were the case.